[Wang Cheng] ip

WangCheng0116 opened the pull request:
# DukePro

*"Your mind is for having ideas, not holding them."* – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind from the burden of remembering tasks. It's:

- Text-based
- Easy to learn
- ~~FAST~~ — SUPER FAST to use

All you need to do is:

1. [ ] [Download it from here.](https://example.com/dukepro)
2. [ ] Double-click it.
3. [ ] Add your tasks.
4. [ ] Let it manage your tasks for you 😉
   
And it is **FREE**!

## Features

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you're a Java programmer, you can also use it to practice Java. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

WangCheng0116 committed:
Rename bot and initialize skeleton

WangCheng0116 committed:
implement echo function

WangCheng0116 committed:
Implement add and list function

WangCheng0116 committed:
Implement mark as done function

WangCheng0116 committed:
creating subclasses of task, consisting of todo, deadline and event

WangCheng0116 committed:
creating TaskParser class and implement function of adding different kinds of tasks

WangCheng0116 committed:
finish A-TextUiTesting by creating input.txt and EXPECTED.txt

WangCheng0116 committed:
creating own InvalidTaskFormatException and hence handle invalid input cases

WangCheng0116 committed:
implement deleting action

WangCheng0116 committed:
expand test cases

WangCheng0116 committed:
Implement TaskReader

WangCheng0116 committed:
Implement TaskWriter

WangCheng0116 committed:
Handle non-existent file case

WangCheng0116 committed:
Merge branch 'branch-Level-7'

WangCheng0116 committed:
Implement reading date feature

WangCheng0116 committed:
Create corresponding packages

WangCheng0116 committed:
Implement prototypes of different commands and parser

WangCheng0116 committed:
Implement DataParser

WangCheng0116 committed:
Implement Storage class and Ui class

WangCheng0116 committed:
Connect each component with main logic

WangCheng0116 committed:
Fix all Java coding standard issues

WangCheng0116 committed:
Merge branch 'add-gradle-support' of https://github.com/WangCheng0116/ip

WangCheng0116 committed:
Fix bugs found when deleting tasks

WangCheng0116 committed:
Add two Junit tests

WangCheng0116 committed:
Fix coding standard

WangCheng0116 committed:
Merge branch 'A-CodingStandard' into A-JavaDoc

WangCheng0116 committed:
Provide JavaDoc for classes in package command

WangCheng0116 committed:
Add JavaDoc to classes in package core, parser and storage

WangCheng0116 committed:
Add JavaDoc to classes within packages tasklist, tasks, ui and test

WangCheng0116 committed:
Merge branch 'A-JavaDoc' into Level-9

WangCheng0116 committed:
Implement finding feature

WangCheng0116 committed:
Merge branch 'Level-9'

WangCheng0116 committed:
Fix indentation issues

WangCheng0116 committed:
Faking branch for auto-grader

WangCheng0116 committed:
Configure checkstyle file and fix checkstyle problems

WangCheng0116 committed:
Merge branch 'A-CheckStyle'

WangCheng0116 committed:
Merge tag 'branch-level-8'

WangCheng0116 committed:
Fix gradle issues

WangCheng0116 committed:
Configure JavaFX and implement a sample usage

WangCheng0116 committed:
Implement UI

WangCheng0116 committed:
Fix bugs in UI

WangCheng0116 committed:
Merge branch 'Level-10'

Singa-pirate commented on src/main/java/commands/AddTodoCommand.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * The regular expression pattern for matching the format of a todo command.
+     */
+    public static final String TODO_PATTERN = "^(todo)\\s+.+";
```
I like the use of a constant field here, and the naming follows coding standard too!

Singa-pirate commented on src/main/java/core/Duke.java:
```diff
@@ -0,0 +1,73 @@
+                    Command c = CommandParser.parse(fullCommand);
+                    c.execute(taskList, ui, storage);
+                }
+            } catch (Exception e) {
```
I suggest you could specify which types of exceptions may be thrown here, or create your own Exception classes

Singa-pirate commented on src/main/java/parser/CommandParser.java:
```diff
@@ -0,0 +1,113 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import commands.AddDeadlineCommand;
```
I love how the imports are explicitly listed and ordered alphabetically. It's really easy to find an item in this list!

Singa-pirate commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+ * methods to add, mark, unmark, delete, and print tasks.
+ */
+public class TaskList {
+    private static ArrayList<Task> ls = DataReader.readTasksFromFile();
```
Perhaps a more detailed name for this variable?

Singa-pirate commented on src/main/java/commands/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,36 @@
+        this.task = task;
+    }
+
+    /**
```
I like how clear the documentations are!

Singa-pirate reviewed (commented):
I really like this pull request, especially the use of regular expressions in input checking!

WangCheng0116 committed:
Improve comments

WangCheng0116 committed:
Add assertions

WangCheng0116 committed:
Merge pull request #1 from WangCheng0116/branch-A-Assertions

Add assertions

WangCheng0116 committed:
Merge pull request #2 from WangCheng0116/branch-A-CodeQuality

Improve comments

WangCheng0116 committed:
Add help function

WangCheng0116 committed:
Add help command to the application

The application lacks a comprehensive help command, leaving users without clear guidance on how to use its features.

To improve user experience and make the application more user-friendly, it's essential to provide users with accessible help documentation.

Introduce a new "help" command that displays helpful information about the application's usage.

The "help" command is added to ensure users can easily access instructions and guidance within the application. It promotes better user understanding and utilization of the software.

This commit includes the implementation of the "help" command, including documentation updates and user instructions.

WangCheng0116 committed:
Fix bugs on handling invalid inputs

WangCheng0116 committed:
Fix coding style

WangCheng0116 committed:
Improve on UI

WangCheng0116 committed:
Improve on UI

WangCheng0116 committed:
Improve UI

WangCheng0116 committed:
Fix UI bugs

WangCheng0116 committed:
Fix bugs in delete command

WangCheng0116 committed:
Update UG

WangCheng0116 committed:
Fix adding deadline bugs

WangCheng0116 committed:
Update README.md

WangCheng0116 committed:
Generate jar file

WangCheng0116 committed:
Fix bugs

WangCheng0116 committed:
Fix Bye command

WangCheng0116 committed:
no message
